apps/platform-app/src/app/[locale]/(profile)/_workflows/profile-evaluation.workflow.ts

```diff
@@ -0,0 +1,152 @@
+import { ToolNode } from '@langchain/langgraph/prebuilt'
+
+import {
+  sendProfileApprovedEmail,
+  sendProfileRejectedEmail,
+} from '@/backend/mailing/mailing.service'
+import { updateProfileById } from '@/backend/profile/profile.service'
+import { getUserById } from '@/backend/user/user.service'
+import { type UserWithProfileAndGH } from '@/backend/user/user.types'
+import { getContextVariable, setContextVariable } from '@langchain/core/context'
+import { tool } from '@langchain/core/tools'
+import { Annotation, type LangGraphRunnableConfig } from '@langchain/langgraph'
+import { z } from 'zod'
+
+const StateAnnotation = Annotation.Root({
+  user: Annotation<UserWithProfileAndGH>(),
+  evaluation: Annotation<string>(),
+})
+
+const evaluationModel = new ChatGroq({
+  model: 'llama-3.1-70b-versatile',
+  temperature: 0.2,
+  streaming: false,
+})
+
+const executionModel = new ChatGroq({
+  model: 'llama3-groq-8b-8192-tool-use-preview',
+  temperature: 0.1,
+})
+
+const acceptProfileTool = tool(
+  async (_, config: LangGraphRunnableConfig) => {
+    const userId = config.configurable?.userId
+    const currentState =
+      getContextVariable<typeof StateAnnotation.State>('currentState')
+
+    if (currentState?.user) {
+      if (currentState?.user.profile) {
+        updateProfileById(currentState.user.profile.id, { state: 'APPROVED' })
+      }
+
+      sendProfileApprovedEmail(
+        currentState?.user.email,
+        currentState?.user.githubDetails?.username || '',
+      )
+    }
+    return `The profile has been accepted. UserId: ${userId} `
+  },
+  {
+    name: 'accept_profile',
+    description: 'Accept user apply and save his profile in our database',
+    schema: z.object({}),
+  },
+)
+
+const rejectProfileTool = tool(
+  async (input, config: LangGraphRunnableConfig) => {
+    const userId = config.configurable?.userId
+    const { reason } = input
+
+    const currentState =
+      getContextVariable<typeof StateAnnotation.State>('currentState')
+
+    if (currentState?.user) {
+      if (currentState.user.profile) {
+        updateProfileById(currentState.user.profile.id, { state: 'REJECTED' })
+      }
+      sendProfileRejectedEmail(currentState.user.email, reason)
+    }
+
+    return `The profile has been rejected. Reason: ${reason}, UserId: ${userId}`
+  },
+  {
+    name: 'reject_profile',
+    description: 'Reject user apply and informing him about reason',
+    schema: z.object({
+      reason: z
+        .string()
+        .describe('The reason why did you rejected user profile'),
+    }),
+  },
+)
+
+const tools = [acceptProfileTool, rejectProfileTool]
+
+import { evaluateProfilePrompt } from './prompts/evaluateProfileNode'
+import { executeDecisionPrompt } from './prompts/executeDecisionNode'
+async function retrieveProfile(
+  state: typeof StateAnnotation.State,
+  config: LangGraphRunnableConfig,
+) {
+  const userId = config.configurable?.userId
+  const fetchedUser = await getUserById(userId)
```

Comment:
**nit:** we should fetch and return profile instead of user. You can include user in profile when fetching it.